File: Swift/robot-return-to-origin.swift
class Solution {
    func judgeCircle(_ moves: String) -> Bool {
        var dict = [Character: Int]()

        for move in moves {
            dict[move] = (dict[move] ?? 0) + 1
        }

        return dict["U"] == dict["D"] && dict["L"] == dict["R"]
    }
}

File: Python/robot-return-to-origin.py
class Solution:
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        d = dict({'U': 0, 'D': 0, 'L': 0, 'R': 0})

        for move in moves:
            d[move] = d[move] + 1

        return d['U'] == d['D'] and d['L'] == d['R']
